src/utilities/math.ts

```diff
@@ -76,37 +76,81 @@ export function getPercentiles(histogram: CARTA.IHistogram, ranks: number[]): nu
     return calculatedPercentiles;
 }
 
-export function scaleValue(x: number, scaling: FrameScaling, alpha: number = 1000, gamma: number = 1.5) {
+function errorFunction(x: number, c: number, x0: number) {
+    const y = Math.exp(c * (x - x0));
+    return y / (y + 1);
+}
+
+export function scaleValue(x: number, scaling: FrameScaling, alpha: number = 1000, gamma: number = 1.5, bias: number = 0, contrast: number = 1,
+    biasContrastMode: boolean = false, smoothedBias: number = 0, smoothedContrast: number = 0) {
+
+    let scaleValue;
     switch (scaling) {
         case FrameScaling.SQUARE:
-            return x * x;
+            scaleValue =  x * x;
+            break;
         case FrameScaling.SQRT:
-            return Math.sqrt(x);
+            scaleValue =  Math.sqrt(x);
+            break;
         case FrameScaling.LOG:
-            return Math.log(alpha * x + 1.0) / Math.log(alpha + 1.0);
+            scaleValue = Math.log(alpha * x + 1.0) / Math.log(alpha + 1.0);
+            break;
         case FrameScaling.POWER:
-            return (Math.pow(alpha, x) - 1.0) / (alpha - 1.0);
+            scaleValue = (Math.pow(alpha, x) - 1.0) / (alpha - 1.0);
+            break;
         case FrameScaling.GAMMA:
-            return Math.pow(x, gamma);
+            scaleValue = Math.pow(x, gamma);
+            break;
         default:
-            return x;
+            scaleValue = x;
+    }
+    if (biasContrastMode) {
```

Comment:
since `biasContrastMode` is a boolean, I think it just be something like `useSmoothBiasContrast`

Reply:
Sure. Will change the naming.